packages/headless/src/plans.ts

```diff
@@ -11,3 +11,66 @@ export function usePlans(): UseQueryResult<IPlan.MappedAttributes[], Error> {
     queryKey: ["find-all-plans"],
   });
 }
+
+export function useCreatePlan({
+  onSuccess,
+  onError,
+}: {
+  onSuccess?(): void;
+  onError?(error: Error): void;
+}) {
+  const atlas = useAtlas();
+  const createPlan = useCallback(
+    (payload: IPlan.CreateApiPayload) => atlas.plan.create(payload),
+    [atlas.plan]
+  );
+  return useMutation({
+    mutationFn: createPlan,
+    mutationKey: ["create-plan"],
+    onSuccess,
+    onError,
+  });
+}
+
+export function useUpdatePlan({
+  id,
+  onSuccess,
+  onError,
+}: {
+  id: number;
+  onSuccess?: Void;
+  onError?(error: Error): void;
+}) {
+  const atlas = useAtlas();
+  const updatePlan = useCallback(
+    (payload: IPlan.UpdateApiPayload) => atlas.plan.update(id, payload),
```

Comment:
The `id` should be passed to the callback and not to the hook.

```suggestion
    ({id, payload}:{id: number, payload: IPlan.UpdateApiPayload) => atlas.plan.update(id, payload),
```